soco/core.py

```diff
@@ -129,6 +130,21 @@ def inner_function(self, *args, **kwargs):
     return inner_function
 
 
+def only_on_soundbars(function):
+    """Decorator that raises NotSupportedException when trying to use soundbar methods on unsupported devices."""
+
+    @wraps(function)
+    def inner_function(self, *args, **kwargs):
+        if not self.is_soundbar:
+            raise NotSupportedException(
+                f"The device is not a soundbar and doesn't support {function.__name__}."
```

Comment:
Using f-strings will break Python 3.5 compatibility. Could you replace with %, please?

Reply:
Done.

Reply:
Yep, I know, but EOL was less than a year ago, and while there's nothing pressurising us to deprecate 3.5 support we might as well continue supporting it.